airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/MapperPipeline.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data
+
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.message.DestinationRecord
+import kotlin.reflect.KClass
+
+class MapperPipeline(
+    inputSchema: AirbyteType,
+    schemaValueMapperPairs:
+        List<Pair<KClass<out AirbyteSchemaMapper>, KClass<out AirbyteValueMapper>>>,
```

Comment:
why does this need to use reflection, rather than just taking instances of the mappers directly?

(or, at least, `Pair<AirbyteSchemaMapper, (meta) -> AirbyteValueMapper>`)

Reply:
Value mappers have to be instantiated with a `DestinationRecord.Meta` that won't be available until record mapping time

Schema mappers don't have the same problem, but ... symmetry?

Reply:
right, that's why I was suggesting taking a function `(meta) -> ValueMapper` - I think that's basically equivalent, just more explicit than the reflection thing?